Spigot-API-Patches/0317-Add-Entity-Visibility-API.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Janmm14 <gitconfig1@janmm14.de>
+Date: Fri, 25 Jun 2021 00:30:54 +0200
+Subject: [PATCH] Add Entity Visibility API
+
+
+diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
+index 46985eaea3d3b00d1dd88c2dd5a2bc53d518c64f..f967983bfb7f3c8318f13ad1eee01589db5e7c13 100644
+--- a/src/main/java/org/bukkit/entity/Entity.java
++++ b/src/main/java/org/bukkit/entity/Entity.java
+@@ -704,4 +704,18 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
+      */
+     public boolean isTicking();
+     // Paper end
++    
++    // Paper start - Entity Visibility API
++    /**
++     * Check if the entity is shown to players by default. Default: true
++     * @return True if the entity is shown to players by default
++     */
++    boolean isShownByDefault();
++
++    /**
++     * Resets to which players this entity is shown and updates the default shown or hidden state
++     * @param shownByDefault True if entity should not be visible to players by default
++     */
++    void resetAndSetShownByDefault(boolean shownByDefault);
```

Comment:
holy fuck I hate this name, maybe don't bundle these operations?

Reply:
The implementation only saves the players per entity which differ from the default state.
This combination saves us from iterating through all players to toggle their existence in the impl WeakHashSet (potentially twice).
I'd prefer to just change the name to something less ugly.